superset/connectors/sqla/models.py

```diff
@@ -656,6 +663,51 @@ def adhoc_metric_to_sqla(self, metric: Dict, cols: Dict) -> Optional[Column]:
 
         return self.make_sqla_column_compatible(sqla_metric, label)
 
+    def process_filters(self, filter, cols):
```

Comment:
any reason we need to take in the cols parameter here?  Might be more self-contained if we just created the col mapping internally since we're able to.  Alternately cols could be optional, but if provided it would give a (tiny?) speedup and make the function more self-documenting.

also naming nitpick - "process_" doesn't really give a huge amount of info as to what this function does - maybe something like "build_and_clauses()"?